zokrates_core/src/imports.rs

```diff
@@ -120,6 +120,25 @@ impl Importer {
                         }
                     }
                 }
+                #[cfg(feature = "ark")]
+                "snark_verify_bls12_377" => {
+                    if T::id() != Bw6_761Field::id() {
+                        return Err(CompileErrorInner::ImportError(
+                            Error::new(format!(
+                                "Embed `snark_verify_bls12_377` must be used with curve `{}`",
+                                Bw6_761Field::name()
+                            ))
+                            .with_pos(Some(pos)),
+                        )
+                        .in_file(location)
+                        .into());
+                    } else {
+                        SymbolDeclaration {
+                            id: symbol.get_alias(),
+                            symbol: Symbol::Flat(FlatEmbed::Verify),
```

Comment:
Shall we rename `FlatEmbed::Verify` to something more precise? `SnarkVerifyBls12377` ?

Reply:
I was thinking about this but I didn't like the naming :/ I guess we can't avoid that